application/senic/nuimo_hub/device_discovery.py

```
@@ -123,16 +147,12 @@ def authenticate(self):
         try:
             payload = json.dumps({"devicetype": self.app_name})
             response = self._request(self.bridge_url, method="POST", payload=payload)
-            if not response:
-                return
-
-            self._username = response["success"]["username"]
-
-            with open(self.state_file_path, "w") as f:
-                json.dump({"devicetype": self.app_name, "username": self._username}, f)
-
+            if response:
+                self.username = response["success"]["username"]
         except UnauthenticatedDeviceError:
-            pass  # consume the exception
+            self.username = None
```

If I set it to `None` then subsequent requests don't have to try reaching bridge to fail and fail fast.